src/group_theory/submonoid/pointwise.lean

```diff
@@ -141,6 +143,14 @@ instance pointwise_central_scalar [mul_distrib_mul_action αᵐᵒᵖ M] [is_cen
   is_central_scalar α (submonoid M) :=
 ⟨λ a S, congr_arg (λ f, S.map f) $ monoid_hom.ext $ by exact op_smul_eq_smul _⟩
 
+/-- The submonoid of powers of a natural number of a commutative submonoid.
+
+This is available as an instance in the `pointwise` locale. -/
+protected def pointwise_has_pow {M : Type*} [comm_monoid M] : has_pow (submonoid M) ℕ :=
+⟨λ S n, S.map $ pow_monoid_hom n⟩
```

Comment:
Can you add the int version here too?

Reply:
I don't know how to do this without assuming that `M` is a `comm_group`, which I just did.

Reply:
I think it's only true with the commutativity assumption anyway.